pkg/prometheus/prometheus.go

```diff
@@ -80,6 +82,56 @@ func getRouter(registry Registry, path string) (*http.ServeMux, error) {
 	return mux, nil
 }
 
+// SetupRegistry returns a fully configured Prometheus registry with metrics based on Kube Linter validations
+func SetupRegistry() (*prometheus.Registry, error) {
+	prom := prometheus.NewRegistry()
+
+	reg, err := validations.GetKubeLinterRegistry()
+	if err != nil {
+		return nil, err
+	}
+
+	checks, err := validations.GetAllNamesFromRegistry(reg)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, checkName := range checks {
+		metric, err := setupMetric(reg, checkName)
+		if err != nil {
+			return nil, fmt.Errorf("unable to create metric for check %s", checkName)
+		}
+
+		if err := prom.Register(metric); err != nil {
```

Comment:
I think we register too many metrics too early here, because if I am not mistaken the config is loaded afterwards (when initializing the `validationEngine`, which creates the metrics again, but never registers them).